### PR TITLE
Add Env#getSupportedMimeTypes()

### PR DESCRIPTION
**Pull request by chrisseaton**
I want to know what languages are supported in a `PolyglotEngine`. The pressing use-case is in JRuby+Truffle where I want to know if I have a JS interpreter available. I'm currently doing this by just trying to eval with `application/javascript` and catching any error as an indication that it isn't available. This patch instead checks what languages are available first.

https://github.com/jruby/jruby/compare/truffle-execjs...truffle-execjs-list-languages#diff-0c9caf96978e9602deec3f1fca2d97a8L68

The workaround before this patch is slow, even if I end up not using the JavaScript interpreter after finding that it is available. This is 3rd party code for the most part so I can't redesign much.

This is a new API, so a 'large' review. @jtulach @chumer @woess

### CODE REVIEW

**Comment by jtulach** [comment]
The test coverage seems to be zero. Please increase it a bit.

File: truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/TruffleLanguage.java [hunk 1]
```diff
@@ -386,6 +386,19 @@ public Object importSymbol(String globalName) {
...
+         * Produces an array of strings for the MIME types supported by this {@link PolyglotEngine}.
+         * There may be multiple MIME types for a single language. Each MIME type will appear only
+         * once, even if multiple language declare it.
+         *
+         * MIME types from this array can be used in methods such as {@link #parse}.
+         *
+         * @return a non-null array of a string for each supported MIME type
+         */
+        public String[] getSupportedMimeTypes() {
+            return API.getSupportedMimeTypes(vm);
```
Comment (jtulach) [inline truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/TruffleLanguage.java]: As far as I understand your use-case a method like
```
public boolean isMimeTypeSupported(String mimeType);
```
would work for you as well. If so, I prefer it to the array of all mime types.

**Comment by chrisseaton** [comment]
I switched from `getSupportedMimeTypes` to `isMimeTypeSupported`.

Where is the best place to test this? I couldn't see a collection of `Env` tests. Do I need to set up a dummy language of my own to get at `Env`?

**Comment by smarr** [comment]
I added a test to `EngineTest` class. But be careful not to use internal APIs in the tests (which are available in that package). @jtulach does not like that :)

**Comment by jtulach** [comment]
There are few testing languages along EngineTest. Feel free to improve them for testing the mime types.

**Comment by jtulach** [comment]
Nice test - no hacks, close to the way people will use this API. Once the formatting is fixed, please merge. I'll work on tagging the release then.

**Comment by chrisseaton** [comment]
I'm not sure how to fix these javadoc warnings. When I run `mx javadoc` locally I get tons of other warnings, and Eclipse doesn't have a problem with the classes and methods I've referenced.

Am I missing something?

https://travis-ci.org/graalvm/truffle/jobs/105453975

**Comment by jtulach** [comment]
The right command is:

mx javadoc

**Pull request merged**